parser/Parser/Parser.py

```diff
@@ -0,0 +1,174 @@
+                else :
+                    # sanitizing
+                    #remove any space characters at end of line
+                    line = line.rstrip()
+                    #replace ’ (typographique quote) to simple quote '
+                    line = line.replace("’", "'")
+                    #replace commas that have no space around by a lower comma character and do the same for escaped comma (preceded by a \) (to distinguish them from commas acting as tags separators)
+                    line = re.sub(r"(\d),(\d)", r"\1‚\2", line)
+                    line = re.sub(r"\\,", "\\‚", line)
+                    #removes parenthesis for roman numeral
+                    line = re.sub(r"\(([ivx]+)\)", r"\1", line, flags=re.I)
+                    yield line
+        yield "" #to end the last entry if not ended
+
+    def normalizing(self,line,lang="default"):
+        """normalize a string depending of the language code lang"""
+        line = unicodedata.normalize("NFC", line)
+        if lang in ['fr','ca','es','it','nl','pt','sk','en']: #removing accent
+            line = unidecode.unidecode(line)
+        if lang not in [ ]: #lower case except if language in list
+            line=line.lower()
+        line = re.sub(r"[\u0000-\u0027\u200b]", "-", line)
+        line = re.sub(r"&\w+;", "-", line)
+        line = re.sub(r"[\s!\"#\$%&'()*+,\/:;<=>?@\[\\\]^_`{\|}~¡¢£¤¥¦§¨©ª«¬®¯°±²³´µ¶·¸¹º»¼½¾¿×ˆ˜–—‘’‚“”„†‡•…‰‹›€™\t]", "-", line)
+        line = re.sub(r"-+", "-", line)
+        line = line.strip("-")
+        return line
+
+
+    def add_line(self,line):
+        """to get a normalized string but keeping the language code "lc:" and the "," separators"""
+        if len(line)<=3:return self.normalizing(line)
+        else:
+            lang=line[:2]
+            new_line=lang+":"
+            for word in line[3:].split(","):
+                new_line+=self.normalizing(word,lang)+','
+            new_line=new_line[:-1]
+            return new_line
+
+    def create_headernode(self,header):
+        query="CREATE (node{id:'__header__'" + (",value:'"+ header + "'" if header else "") +"})"
+        self.session.run(query)
+
+    def create_node(self,data):
+        id_query = " CREATE (n{id:'" + data['id'] + "'}) SET n.block_index=" + str(self.block_index) + ", n.block_subindex=0, n.preceding_lines="+str(data['comment'])
+        if data['value'] : entry_query = " SET n.value='" + str(data['value']) + "'"
+        else : entry_query = " SET n.tags=\"" + str(data['tags']) + "\" ,n.tagsid=" + str(data['tagsid']) + " ,n.parents=" + str(data['parent_tag']) + " ,n.properties=" + str(data['properties'])
+        query=id_query+entry_query
+        # print(query)
+        self.session.run(query)
+
+    def reinitialize_data(self):
+        data = { "id":'',
+        "comment":[],
+        "parent_tag":[],
+        "tags":{},
+        "tagsid":[],
+        "properties":[],
+        "value":'',
+        }
```

Comment:
Do include some indentation and spaces while declaring dictionaries, since it makes the dictionary more readable :) 
```suggestion
        data = { 
                "id" : '',
                "comment" : [],
                "parent_tag" : [],
                "tags" : {},
                "tagsid" : [],
                "properties" : [],
                "value" : '',
        }
```